### PR TITLE
YALB-419: Add Example Form to Storybook

### PR DESCRIPTION
**Pull request by ModulesUnraveled**
### Relates to the following PRs:
- https://github.com/yalesites-org/atomic/pull/6
- https://github.com/yalesites-org/yalesites_profile/pull/22

### Description of work
- Creates a generic "Embed" component
- Embeds an example Qualtrics form and displays it on a page

### Testing Link(s)
- [ ] Navigate to the [Embed Story](https://deploy-preview-75--dev-component-library-twig.netlify.app/?path=/story/molecules-embed--embed)
- [ ] Navigate to the [Qualtrics Page Example](https://deploy-preview-75--dev-component-library-twig.netlify.app/?path=/story/page-examples-page-examples--qualtrics-page)

### Functional Review Steps
- [ ] Verify you can install the component with the CLI
- [ ] Verify the form appears in the embed

### Design Review
- [ ] Verify the designs match the [Figma Designs](https://www.figma.com/file/OVfmCW5s1gI7xnrM8ibX2y/UI-Kit-Alpha-%5BYaleSites%5D)

### Accessibility Review
- [ ] Verify the component meets Accessibility requirements

### CODE REVIEW

**Review by kara-franco — CHANGES_REQUESTED** [review]
Two question/thoughts below: 

The Embed title name should be required in Drupal. This will ensure that all embeds have accessible titles (helps screen reader users). 

Can we or should we name this component "Form Embed" to differentiate between the "Media Embed" which will have video/audio? 

Reference shot (title attribute): 
Screen Shot 2022-04-21 at 1 27 57 PM

**Comment by ModulesUnraveled** [comment]
@kara-franco I agree. I'm assuming that we'll accept (and require) both the form (page) URL, and a title for the form (page). That way the title attribute will always be provided.

And I'm not sure about naming it "Form Embed"... there's nothing form-specific about it. It's actually just a generic iframe component at this point. And I don't know if we'll call the media ones "Media Embed", or if they'll just be called "Media" - or even "Video", "Audio", etc.

I think I'd be more inclined to call this component "Iframe" if anything, since that's really what it is. It doesn't support any other type of "embeds" that people might be used to.

**Comment by kara-franco** [comment]
> @kara-franco I agree. I'm assuming that we'll accept (and require) both the form (page) URL, and a title for the form (page). That way the title attribute will always be provided.
> 
> And I'm not sure about naming it "Form Embed"... there's nothing form-specific about it. It's actually just a generic iframe component at this point. And I don't know if we'll call the media ones "Media Embed", or if they'll just be called "Media" - or even "Video", "Audio", etc.
> 
> I think I'd be more inclined to call this component "Iframe" if anything, since that's really what it is. It doesn't support any other type of "embeds" that people might be used to.

@ModulesUnraveled 
**Requiring page URL and title-** perfect!! 
**Name of Embed component-** that makes good sense! I think my gut reaction was if a developer came to Storybook looking for a form example- it may be hard to find. But, the way you are explaining it makes more sense. And, we will eventually have a Form component example (Subscribe or Contact Us) to reference. 

Thank you!!!

**Review by kara-franco — APPROVED** [review]

**Comment by ahitson** [comment]
Is there a way to have the embed live in the space in full and not in a scrollable frame?

**Comment by ModulesUnraveled** [comment]
@ahitson Unfortunately, no. The only way to embed Qualtics forms is in an iframe, and iframes don't have the ability (yet) to dynamically change size based on the content. So we just have to give it a fixed size (in this case, we're doing 70% of the screen height, so it's a little responsive, but still not perfect). Unfortunately, it's as good as we can get 😞

**Review by ahitson — APPROVED** [review]

**Review by callinmullaney — CHANGES_REQUESTED** [review]
@ModulesUnraveled This breaks down to mobile surprisingly well. See my code comment regarding the height... not sure if we should do something about it given this is iframed content that we have limited options with.

The border style between Chrome/FF is different for `<iframe>`. I think the iframe is just inheriting reset styles that should either be `border:0` or one of the existing border tokens explicitly. 
Screenshot from 2022-04-26 14-47-49

**Comment by callinmullaney** [comment]
Wow that screenshot sucks :shrug:

File: components/02-molecules/embed/_embed.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.embed__iframe {
+  height: 70vh;
```
[the remark this answers is not in the document]
Comment (ModulesUnraveled) [reply components/02-molecules/embed/_embed.scss:2]: I think I'd rather not for now. We don't know what the content is going to be, and we can always adjust this once we get real forms starting to get entered. I'd hate to restrict it for now, and then find out that all of the forms are enormous, and we arbitrarily restricted the height, causing issues.

I'd like to leave it as flexible as possible for now, and then iterate if needed.

We might even end up giving people a "form size" option with "small", "medium", and "large" options that defines the max height... who knows?

**Comment by callinmullaney** [comment]
@ModulesUnraveled one thing I forgot to mention is I think the iframe should be set as `display: block` looking at the page example there is a small gap between the iframe and wrapping `embed__inner`. See:
Screen Shot 2022-04-26 at 4 01 04 PM

File: components/02-molecules/embed/embed.twig [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ # Available Variables:
+ # - embed__src
+ # - embed__title
+ #}
+
+{% set embed__base_class = 'embed' %}
+
+{% set embed__attributes = {
+  'data-component-width': 'max',
+  class: bem(embed__base_class),
```
[the remark this answers is not in the document]
Comment (ModulesUnraveled) [reply components/02-molecules/embed/embed.twig:11]: This is a prettier formatting thing. I think it's interpreting this as YML or something, which expects keys to be not in parentheses UNLESS the key has a dash in it... which we do... then the key MUST be in quotes.


UPDATE: I totally lied... the keys with dashes MUST have quotes, but the ones without can... and prettier isn't formatting twig... so I wonder if it's just something I brought over from working with the tokens and stuff...

...

I don't know if this was a conscious choice or just an artifact of development that just stuck... `class` works with, and without the quotes... So I think I'd rather have the quotes to keep things consistent (as far as color highlighting goes in my editor) but I also don't think we necessarily need to refactor every other component just to add the quotes...

I'll go ahead and add the quotes here, and if/when we update others make sure to add them in along the way.

**Comment by ModulesUnraveled** [comment]
@callinmullaney I think I've addressed all of your feedback. I got Ashley to weigh in on the border, so we set that to be consistent across browsers.

File: components/02-molecules/embed/_embed.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (callinmullaney) [reply components/02-molecules/embed/_embed.scss:7]: :+1:

File: components/02-molecules/embed/embed.twig (hunk 2 above)
[the remark this answers is not in the document]
Comment (callinmullaney) [reply components/02-molecules/embed/embed.twig]: Sounds good. I figured it wasn't a big deal either but.

**Review by callinmullaney — APPROVED** [review]
Changes look good! :shipit:

**Pull request merged**